driver/main.jl

```
@@ -265,13 +265,20 @@ function run(sim::Simulation1d; time_run = true)
     sim.skip_io || open_files(sim.Stats) # #removeVarsHack
     (prob, alg, kwargs) = solve_args(sim)
     integrator = ODE.init(prob, alg; kwargs...)
-    if time_run
-        sol = @timev ODE.solve!(integrator)
-    else
-        sol = ODE.solve!(integrator)
+
+    local sol
+    try
+        if time_run
+            sol = @timev ODE.solve!(integrator)
+        else
+            sol = ODE.solve!(integrator)
+        end
+    catch e
+        return :simulation_crashed
```

Is the `finally` reached if you return here?

---

finally will be called before return.  it would be more natural to throw an exception though (and wrap the original exception)?

---

Yep:
```julia
julia> function foo()
           try
               error("oops")
               return 2
           catch
               return 1
           finally
               println("finally")
           end
       end
foo (generic function with 1 method)

julia> foo()
finally
1
```

---

> it would be more natural to throw an exception though (and wrap the original exception)?

Maybe you're right, but one advantage I've realized about working with the return code is that, instead of error-ing out, we can continue to the plotting scripts and (try) to plot results of the crashed solution and test which variables are blowing up.

---

you don't have to error out though, you could capture the exception at a higher level (log it?) and then continue with the control flow.

---

Ahhh, I see, yeah, we can do that

---

I was thinking about this a bit more, if we returned an exception, wouldn't that make `main` type-unstable? I think, for now, just adding a warning to the log and returning a symbol is a significant memory management improvement that we should get merged, so let's move forward and we can open an issue about better exception handling.